src/ThreadPool.cc

```diff
@@ -262,6 +285,11 @@ void ThreadPool::perform_work()
             running_task_names_.erase(running_task_names_.begin() + idx);
         }
     }
+    // Mark ourselves dead before we stop executing
+    for (auto& t : threads_) {
+        if (std::this_thread::get_id() == t.t.get_id())
```

Comment:
Isn't this equal to [`not joinable()`](https://en.cppreference.com/w/cpp/thread/thread/joinable)?

Reply:
I guess I don't get it.

Do you mean `t.running` - because when it stopped running it becomes joinable?
I can't really remember, just in the dark, that joinable is not always the same as running.
But reading the docs it seems you are right.
\*pondering\*

Yes, I'll add a commit that simplifies this.

Reply:
Ok, my previous comment was of course completely wrong.

We need `Thread` that contains the `std::thread` and the `bool running`, because just from the `std::thread` you can not determine if the thread is doing something or not. From cppref:

![image](https://github.com/gul-cpp/gul14/assets/16012374/a4fe87c5-bf48-4c0d-a5bc-2c4ddc47e199)

The thread is joinable immediately after it has been started and is still joinable if it stopped working, until someone joined it. So you can not determine if the thread is executing code or finished executing but waiting to join.

But I guess I just did not understand your question.

Maybe you mean this comment from cppref:

![image](https://github.com/gul-cpp/gul14/assets/16012374/87342a36-10d4-48c4-825a-5b3931ac745e)

So you maybe suggest that we use `joinable()` here instead of the explicit id comparison?

Unfortunately the comment on cppref seems wrong. It just does not work this way and when you compare the two flags you see it controadicts it and we must use the explicit id check.

Here my code / test. Note how this contradicts that enigmatic sentence on cppref.

![image](https://github.com/gul-cpp/gul14/assets/16012374/909acf0f-8228-44ad-bc78-975481365049)

----

So, answering your original question, no, it is not equal; and we must use the explicit id comparsion.